src/app/external-cluster-wizard/steps/external-cluster/provider/aks/component.ts

```diff
@@ -0,0 +1,192 @@
+  templateUrl: './template.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => AKSClusterSettingsComponent),
+      multi: true,
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => AKSClusterSettingsComponent),
+      multi: true,
+    },
+  ],
+})
+export class AKSClusterSettingsComponent
+  extends StepBase
+  implements OnInit, OnDestroy, ControlValueAccessor, Validator
+{
+  readonly Controls = Controls;
+  readonly ErrorType = ErrorType;
+  readonly AUTOSCALING_MIN_VALUE = 1;
+  readonly AUTOSCALING_MAX_VALUE = 1000;
+  isLoadingVmSizes: boolean;
+  vmSizes: string[] = [];
+  @Input() projectID: string;
+  private readonly _debounceTime = 500;
+
+  constructor(
+    private readonly _builder: FormBuilder,
+    private readonly _externalClusterService: ExternalClusterService,
+    private readonly _nameGenerator: NameGeneratorService
+  ) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this._initForm();
+    this._initSubscriptions();
+  }
+
+  ngOnDestroy(): void {
+    this.reset();
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  generateName(): void {
+    this.form.get(Controls.Name).setValue(this._nameGenerator.generateName());
+  }
+
+  onEnableAutoScalingChange(evt: MatCheckboxChange) {
+    if (!evt.checked) {
+      this.form.patchValue({
+        [Controls.MaxCount]: null,
+        [Controls.MinCount]: null,
+      });
+    }
+  }
+
+  private _initForm() {
+    const MIN_COUNT_DEFAULT_VALUE = 1;
+    const MAX_COUNT_DEFAULT_VALUE = 5;
+    this.form = this._builder.group({
+      [Controls.Name]: this._builder.control('', Validators.required),
+      [Controls.Location]: this._builder.control('', Validators.required),
+      [Controls.NodeResourceGroup]: this._builder.control('', Validators.required),
+      [Controls.KubernetesVersion]: this._builder.control('', Validators.required),
+      [Controls.NodePoolName]: this._builder.control('', Validators.required),
+      [Controls.Count]: this._builder.control(1, Validators.required),
+      [Controls.VmSize]: this._builder.control('', Validators.required),
+      [Controls.Mode]: this._builder.control('System'),
+      [Controls.EnableAutoScaling]: this._builder.control(true),
+      [Controls.MaxCount]: this._builder.control(MAX_COUNT_DEFAULT_VALUE, [
+        Validators.required,
+        Validators.max(this.AUTOSCALING_MAX_VALUE),
+      ]),
+      [Controls.MinCount]: this._builder.control(MIN_COUNT_DEFAULT_VALUE, [
+        Validators.required,
+        Validators.min(this.AUTOSCALING_MIN_VALUE),
+      ]),
+    });
+
+    this.form.get(Controls.Mode).disable();
+  }
+
+  private _initSubscriptions() {
+    this.form.valueChanges.pipe(takeUntil(this._unsubscribe)).subscribe(_ => {
+      this._updateExternalClusterModel();
+      this._externalClusterService.isClusterDetailsStepValid = this.form.valid;
+    });
+
+    this.form
+      .get(Controls.Location)
```

Comment:
Perhaps we should use `this.control(Controls.Location)` instead of `this.form.get`?